rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/NodeSourcePluginManager.java

```diff
@@ -0,0 +1,171 @@
+        String newValue;
+        String oldValue;
+
+        for (int newValueIndex = 0; newValueIndex < newParameters.length; newValueIndex++) {
+
+            oldValueIndex = 0;
+
+            for (ConfigurableField configurableField : configurableFields) {
+
+                // we know if two parameters are comparable thanks to the
+                // order in which they appear in the parameter list
+                if (oldValueIndex == newValueIndex) {
+
+                    if (configurableField.getMeta().credential() || configurableField.getMeta().fileBrowser() ||
+                        configurableField.getMeta().password()) {
+                        newValue = new String((byte[]) newParameters[newValueIndex]);
+                        oldValue = new String((byte[]) oldParameters.get(newValueIndex));
+                    } else {
+                        newValue = (String) newParameters[newValueIndex];
+                        oldValue = (String) oldParameters.get(newValueIndex);
+                    }
+
+                    this.updateDynamicParameterIfNotEqual(newParameters,
+                                                          parametersWithDynamicUpdated,
+                                                          newValue,
+                                                          oldValue,
+                                                          newValueIndex,
+                                                          configurableField);
+                }
+
+                oldValueIndex++;
+            }
+        }
+
+        return parametersWithDynamicUpdated;
+    }
+
+    public Collection<PluginDescriptor> getPluginsDescriptor(Collection<Class<?>> plugins) {
+        Collection<PluginDescriptor> descriptors = new ArrayList<>(plugins.size());
+        for (Class<?> cls : plugins) {
+            Map<String, String> defaultValues = new HashMap<>();
+            descriptors.add(new PluginDescriptor(cls, defaultValues));
+        }
+        return descriptors;
+    }
+
+    public PluginDescriptor getPolicyPluginDescriptor(String nodeSourceName,
+            NodeSourceDescriptor nodeSourceDescriptor) {
+        Class<NodeSourcePolicy> policyClass = this.getNodeSourcePolicyClassOrFail(nodeSourceName, nodeSourceDescriptor);
+
+        return new PluginDescriptor(policyClass, nodeSourceDescriptor.getPolicyParameters());
+    }
+
+    public PluginDescriptor getInfrastructurePluginDescriptor(String nodeSourceName,
+            NodeSourceDescriptor nodeSourceDescriptor) {
+        Class<InfrastructureManager> infrastructureClass = this.getInfrastructureManagerClassOrFail(nodeSourceName,
+                                                                                                    nodeSourceDescriptor);
+
+        return new PluginDescriptor(infrastructureClass, nodeSourceDescriptor.getInfrastructureParameters());
+    }
+
+    private void updateDynamicParameterIfNotEqual(Object[] newParameters,
```

Comment:
Do we really need 'newParameters'? newValue is not enough?

Reply:
Yes you are right, applied.